<launch>
    <node name="ORBPOSE_FETCH" pkg="ORB_SLAM3" type="ORBPOSE_FETCH" output="screen">
        <!-- for EuRoC dataset -->
        <remap from="/camera/rgb/image_raw"                 to="/head_camera/rgb/image_raw"/>
        <remap from="/camera/depth_registered/image_raw"    to="/head_camera/depth/image_raw"/>

        <!-- Parameters for original ORB-SLAM3 -->
        <param name="voc_file"      type="string"   value="$(find ORB_SLAM3)/../../../Vocabulary/ORBvoc.txt" />
        <param name="settings_file" type="string"   value="$(find ORB_SLAM3)/../../../Examples/RGB-D/fetch.yaml" />
    
        <!-- Parameters for ROS -->
        <param name="map_frame_id"  type="string"   value="world" />
        <param name="pose_frame_id" type="string"   value="camera" />
    </node>
    
    <node pkg="hector_trajectory_server" type="hector_trajectory_server" name="trajectory_server_orb_slam3" output="screen" ns="ZED_ORBSLAM3" >
        <param name="/target_frame_name" value="/world" />
        <param name="/source_frame_name" value="/camera" />
        <param name="/trajectory_update_rate" value="20.0" />
        <param name="/trajectory_publish_rate" value="20.0" />
    </node>
</launch>